internal/ir/xds.go

```diff
@@ -612,7 +612,9 @@ type TCPListener struct {
 	Port uint32
 	// TLS information required for TLS Passthrough, If provided, incoming
 	// connections' server names are inspected and routed to backends accordingly.
-	TLS *TLSInspectorConfig
+	TLSInspectorConfig *TLSInspectorConfig
```

Comment:
thoughts on making this
```
tls:
  passthrough:
  terminate:
```
?

Reply:
I feel in future we could use SNIs to route traffic in TCP terminate mode as well? Wdyt? Istio does it on the basis of gateway hostname. However at the moment it does not make sense as TCPRoute does not have any SNI based routing. Maybe I am missing something. Wdyt?

Reply:
that case is handled in the `HTTPListener` https://github.com/envoyproxy/gateway/blob/334422166f208a03440fcec1534c8fe4a2aab3ef/internal/ir/xds.go#L117, the Gateway.Listener.Hostname API field internally performs a SNI match at the xds Listener level